webapp/shop/advantage/views.py

```diff
@@ -34,6 +35,121 @@
 )
 
 
+@shop_decorator(area="advantage", permission="user", response="html")
+def get_activate_view(ua_contracts_api, advantage_mapper, **kwargs):
+    account = None
+    user = user_info(flask.session)
+    name = user["fullname"]
+    try:
+        account = advantage_mapper.get_purchase_account("canonical-ua")
+        if account.type == "paid":
+            return flask.render_template(
+                "pro/activate.html",
+                name=name,
+                needs_paid_account_created=False,
+            )
+    except UAContractsUserHasNoAccount:
+        return flask.render_template(
+            "pro/activate.html", name=name, needs_paid_account_created=True
+        )
+    except AccessForbiddenError:
+        return flask.render_template("account/forbidden.html")
+    else:
+        flask.render_template("account/forbidden.html")
+
+
+@shop_decorator(area="advantage", permission="user", response="html")
+def pro_activate_activation_key(ua_contracts_api, advantage_mapper, **kwargs):
+    marketplace = kwargs.get("marketplace", "canonical-ua")
```

Comment:
For my education, is there any chance we will use this page for activating CUBE keys? They are in the `canonical-cube` marketplace, so I was wondering if this may actually be a design upfront.

Again, I am just an observer here and don't have any say, so feel free to overwrite.